.github/workflows/CI_build.yml

```
@@ -7,49 +7,82 @@ on:
+      - name: Python info
+        run: |
+          which python
+          python --version
+      - name: Build package and create dev environment
+        run: |
+          python -m pip install --upgrade pip
+          python setup.py build
+          pip install -e .[dev]
+      - name: Test with coverage
+        run: |
+          pytest --cov --cov-report term --cov-report xml --junitxml=xunit-result.xml
+      - name: Correct coverage paths
+        run: sed -i "s+$PWD/++g" coverage.xml
+      - name: Check style against standards using prospector
+        shell: bash -l {0}
+        run: prospector -o grouped -o pylint:pylint-report.txt
+      - name: Check whether import statements are used consistently
+        shell: bash -l {0}
+        run: isort --check-only --diff --conda-env spec2vec-dev --recursive --wrap-length 79 --lines-after-imports 2 --force-single-line --no-lines-before FUTURE --no-lines-before STDLIB --no-lines-before THIRDPARTY --no-lines-before FIRSTPARTY --no-lines-before LOCALFOLDER spec2vec/ tests/ integration-tests/
+      - name: SonarCloud Scan
+        if: github.repository == 'iomega/spec2vec'
+        uses: sonarsource/sonarcloud-github-action@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
+  build_pypi:
+    name: Pypi and documentation build / python-${{ matrix.python-version }} / ${{ matrix.os }}
     runs-on: ${{ matrix.os }}
+    needs: first_check
     strategy:
       fail-fast: false
       matrix:
         os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
-        python-version: ["3.8"]
-        include:
-          - os: 'ubuntu-latest'
-            python-version: "3.7"
+        python-version: ['3.7', '3.8']
+        exclude:
+          # already tested in first_check job
+          - python-version: 3.8
+            os: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
         with:
-          fetch-depth: "0"
-      - uses: goanpeca/setup-miniconda@v1
-        with:
-          activate-environment: spec2vec-dev
-          auto-update-conda: true
-          environment-file: conda/environment-dev.yml
           python-version: ${{ matrix.python-version }}
-      - name: Show conda config
-        shell: bash -l {0}
-        run: |
-          conda info
-          conda list
-          conda config --show-sources
-          conda config --show
       - name: Python info
-        shell: bash -l {0}
         run: |
           which python
           python --version
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+      - name: Build
+        run: |
+          python setup.py build
+      - name: Show pip list
+        run: |
+          pip list
+      - name: Test
+        run: |
+          pip install -e .[dev]
+          pytest
```

Can you split this step so it's easier to read logs
```suggestion
      - name: Install development dependencies
        run: |
          pip install -e .[dev]
      - name: Test
        run: |
          pytest
```

---

Yes, good idea.